### PR TITLE
grubby: use dedicated installkernel package

### LINKED ISSUE #5493: Installkernel file conflict between installkernel and grubby packages
It is not possible to install both grubby and installkernel packages on the same Mariner environment due to a conflict with the `/usr/sbin/intallkernel` script.

Error message: ` file /usr/sbin/installkernel from install of installkernel-1.0.0-2.cm2.noarch conflicts with file from package grubby-8.40-44.cm2.x86_64`

Full log:
```
$ sudo dnf install build-essential
Last metadata expiration check: 0:00:41 ago on Thu May 11 20:02:12 2023.
Dependencies resolved.
=============================================================================================================================================================================================================================================
 Package                                                    Architecture                                    Version                                                     Repository                                                      Size
=============================================================================================================================================================================================================================================
Installing:
 build-essential                                            x86_64                                          0.1-5.cm2                                                   mariner-official-base                                          7.1 k
Installing dependencies:
 autoconf                                                   noarch                                          2.71-3.cm2                                                  mariner-official-base                                          602 k
 automake                                                   noarch                                          1.16.5-1.cm2                                                mariner-official-base                                          503 k
 binutils                                                   x86_64                                          2.37-5.cm2                                                  mariner-official-base                                          8.9 M
 binutils-devel                                             x86_64                                          2.37-5.cm2                                                  mariner-official-base                                          1.4 M
 bison                                                      x86_64                                          3.7.6-1.cm2                                                 mariner-official-base                                          776 k
 diffutils                                                  x86_64                                          3.8-1.cm2                                                   mariner-official-base                                          510 k
 docbook-dtd-xml                                            noarch                                          4.5-11.cm2                                                  mariner-official-base                                           91 k
 docbook-style-xsl                                          noarch                                          1.79.1-13.cm2                                               mariner-official-base                                           20 M
 glibc-devel                                                x86_64                                          2.35-3.cm2                                                  mariner-official-base                                          534 k
 gtk-doc                                                    noarch                                          1.33.2-1.cm2                                                mariner-official-base                                          817 k
 installkernel                                              noarch                                          1.0.0-2.cm2                                                 mariner-official-base                                           15 k
 kernel-headers                                             noarch                                          5.15.107.1-2.cm2                                            mariner-official-base                                          1.6 M
 libxml2-devel                                              x86_64                                          2.10.3-1.cm2                                                mariner-official-base                                           97 k
 libxslt                                                    x86_64                                          1.1.34-7.cm2                                                mariner-official-base                                          187 k
 m4                                                         x86_64                                          1.4.19-1.cm2                                                mariner-official-base                                          208 k
 patch                                                      x86_64                                          2.7.6-7.cm2                                                 mariner-official-base                                          131 k
 python3-lxml                                               x86_64                                          4.9.1-1.cm2                                                 mariner-official-base                                          1.7 M
 python3-pygments                                           noarch                                          2.4.2-7.cm2                                                 mariner-official-base                                          1.9 M

Transaction Summary
=============================================================================================================================================================================================================================================
Install  19 Packages

Total size: 40 M
Installed size: 103 M
Is this ok [y/N]: y
Downloading Packages:
[SKIPPED] autoconf-2.71-3.cm2.noarch.rpm: Already downloaded
[SKIPPED] automake-1.16.5-1.cm2.noarch.rpm: Already downloaded
[SKIPPED] binutils-2.37-5.cm2.x86_64.rpm: Already downloaded
[SKIPPED] binutils-devel-2.37-5.cm2.x86_64.rpm: Already downloaded
[SKIPPED] bison-3.7.6-1.cm2.x86_64.rpm: Already downloaded
[SKIPPED] build-essential-0.1-5.cm2.x86_64.rpm: Already downloaded
[SKIPPED] diffutils-3.8-1.cm2.x86_64.rpm: Already downloaded
[SKIPPED] docbook-dtd-xml-4.5-11.cm2.noarch.rpm: Already downloaded
[SKIPPED] docbook-style-xsl-1.79.1-13.cm2.noarch.rpm: Already downloaded
[SKIPPED] glibc-devel-2.35-3.cm2.x86_64.rpm: Already downloaded
[SKIPPED] gtk-doc-1.33.2-1.cm2.noarch.rpm: Already downloaded
[SKIPPED] installkernel-1.0.0-2.cm2.noarch.rpm: Already downloaded
[SKIPPED] kernel-headers-5.15.107.1-2.cm2.noarch.rpm: Already downloaded
[SKIPPED] libxml2-devel-2.10.3-1.cm2.x86_64.rpm: Already downloaded
[SKIPPED] libxslt-1.1.34-7.cm2.x86_64.rpm: Already downloaded
[SKIPPED] m4-1.4.19-1.cm2.x86_64.rpm: Already downloaded
[SKIPPED] patch-2.7.6-7.cm2.x86_64.rpm: Already downloaded
[SKIPPED] python3-lxml-4.9.1-1.cm2.x86_64.rpm: Already downloaded
[SKIPPED] python3-pygments-2.4.2-7.cm2.noarch.rpm: Already downloaded
Running transaction check
Transaction check succeeded.
Running transaction test
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Transaction test error:
  file /usr/sbin/installkernel from install of installkernel-1.0.0-2.cm2.noarch conflicts with file from package grubby-8.40-44.cm2.x86_64
```

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

There is a separate dedicated installkernel RPM package that supplies the proper installkernel script for our distribution.

Grubby also supplied a similar script, but it does not support our distribution's specific quirks. Yet because grubby supplies this script at the same location, there are package conflicts when one attempts to install both the grubby package and installkernel package together.

So remove the grubby package's version of the installkernel script and just rely on the dedicated package's version, which is now a package requires for the grubby package.

Fixes #5493 

###### Change Log  

- Remove installkernel script and relevant components from grubby package
- Add package requires for the installkernel package to the grubby spec

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- https://microsoft.visualstudio.com/OS/_workitems/edit/44564856
- #5493 

###### Test Methodology

Builds:
   - x86_64: [359161](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=359161&view=results)
   - ARM64: [359160](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=359160&view=results)

Tested the following:
   - File conflict with installkernel is resolved
   - Main grubby application can still update the grubenv
   - Package update from old grubby to new grubby package functions as expected
   - Package update also installs installkernel package

### CODE REVIEW

File: SPECS/grubby/grubby.spec [hunk 1]
```diff
@@ -69,15 +67,15 @@ users with existing grubby users.
 %install
 %make_install mandir=%{_mandir} sbindir=%{_sbindir} libexecdir=%{_libexecdir}
 
-mkdir -p %{buildroot}%{_libexecdir}/{grubby,installkernel}/ %{buildroot}%{_sbindir}/
+# We do not supply the installkernel script from grubby. Instead users should
```
Comment (Camelron) [inline SPECS/grubby/grubby.spec:70]: Should we add a `Requires: installkernel` so that make install still works properly?

**Review by Alberto-Perez-Guevara — APPROVED** [review]
Changes look good to me, quick question what is the context/reason to remove install kernel?

in the comments it mentions:
#.... Instead users should
# install it from the dedicated installkernel package supplied by our distro. 

but for my education I would like to understand why is this the case, other than that I have no problem with the changes here

**Review by Camelron — APPROVED** [review]
Seems reasonable to me.

File: SPECS/grubby/grubby.spec (hunk 1 above)
Comment (christopherco) [reply SPECS/grubby/grubby.spec:70]: I like this suggestion. Let me try it out.

**Comment by christopherco** [comment]
> Changes look good to me, quick question what is the context/reason to remove install kernel?
> 
> in the comments it mentions: #.... Instead users should
> 
> # install it from the dedicated installkernel package supplied by our distro.
> but for my education I would like to understand why is this the case, other than that I have no problem with the changes here

I have opened a github issue to help describe this issue #5493

**Review by Camelron — APPROVED** [review]
lgtm!

**Pull request merged**